taigaio/taiga-back
# US #632: Task #762: change settings on delete closed status if is setted in gitlab module config

![](https://media.giphy.com/media/J5e7VdW7oo8XTxI13G/giphy.gif)


There are two commits here:

- the first one is a minor refactor and fix MoveOnDestroy mixin to check permision over the two items (deleted and the new destination).
- the second one, the main, change close status in gitlab module config if the current one is deleted.

### How To Test:

 - [x] Prerequisites
   - [x] Connect a project (project with id 1) with gitlab following this steps
- [x] Test 0: Review
  - [x] Pass the integration test.
  - [x] Open python shell with `./manage.py shell` and set custom settings to gitlab module integration
    ```python
    from taiga.projects.models import *

    p = Project.objects.get(id=1)

    s = ProjectModulesConfig.objects.create( 
        project=p, 
        config={ 
            "gitlab": { 
                "close_status": p.issue_statuses.filter(slug="closed").first().id
            } 
        } 
    )

    s.config  # to see the close_status id = 3
    ```
  - [x] Destroy the status Ready for Test [id=3] (and move to Close [id=4]) in other terminal with:
    ```bash
    TOKEN=$(
            curl -X POST \
                -H "Content-Type: application/json" \
                -d '{"type": "normal", "username": "admin", "password": "123123"}' \
                http://localhost:8000/api/v1/auth | jq -r '.auth_token'
            )
    ```
    ```bash
    curl -v -X DELETE \
        -H "Content-Type: application/json" \
        -H "Authorization: Bearer $TOKEN" \
        http://localhost:8000/api/v1/issue-statuses/3?moveTo=4
    ```
  - [x] Check that gitlab module config has been changed, in the python shell:
    ```python
    s.refresh_from_db() 
    s.config  # to see the new close_status id = 4
    ```
- [x] Move [Task #762](https://tree.taiga.io/project/taiganext/task/762) to `ready-for-qa`
